backend/internal/booster_gen/boosters.go

```diff
@@ -0,0 +1,167 @@
+package boostergen
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"slices"
+	"strings"
+
+	scryfallapi "github.com/BlueMonday/go-scryfall"
+	"github.com/joaquinleonarg/wdml_mtg/backend/domain"
+	"github.com/joaquinleonarg/wdml_mtg/backend/pkg/scryfall"
+	"github.com/rs/zerolog/log"
+)
+
+type BoosterData struct {
+	CardCount  int      `json:"cardCount"`
+	Sets       []string `json:"sets"`
+	DefaultSet string   `json:"defaultSet"`
+	Slots      []struct {
+		Options []Option `json:"options"`
+		Count   int      `json:"count"`
+	} `json:"slots"`
+}
+
+type Option struct {
+	Rarity       string   `json:"rarity"`
+	Weight       int      `json:"weight"`
+	Types        []string `json:"types"`
+	SkippedTypes []string `json:"skippedTypes"`
+	Layout       string   `json:"layout"`
+	Set          string   `json:"set"`
+}
+
+func CheckIfBoosterExists(setCode string) (bool, error) {
+	path, err := filepath.Abs("./internal/booster_gen/sets/" + strings.ToLower(setCode) + ".json")
+	if err != nil {
+		return false, err
+	}
+	_, err = os.ReadFile(path)
+	if err != nil {
+		return false, fmt.Errorf("json read error")
+	}
+	return true, nil
+}
+
+func GenerateBoosterFromJson(setCode string) ([]domain.CardData, error) {
```

Comment:
Might be worth it to refactor this function since it got a bit out of hand already, and we probably are only gonna expand on it in the future.